src/intTest/java/ca/nrc/cadc/UserStorageBrowserPage.java

```diff
@@ -311,39 +323,124 @@ public void deleteFolder(String foldername) throws Exception
+            if (currentPermission != null) {
+                // clear checkbox
+                // read group field should be enabled
+                click(permissionCheckbox);
+            }
+            sendKeys(groupInput, newGroup);
+        }
+
+        // Wait for autocomplete to populate & be displayed or this
+        // may fail - autocomplete dropdown doesn't display consistently
+        // after sendkeys so isn't proving to be reliable to check that element
+        // is visible. :(
+        Thread.sleep(4000);
+
         clickButton(SAVE);
 
-        // confirm folder delete
-        if (isJqiMsgShowing(SUCCESSFUL))
+        confirmJqiMsg(SUCCESSFUL);
+    }
+
+    /**
+     * Convenience function to click through most of the impromptu .prompt
+     * confirmation patterns.
+     * @param messageType
+     * @throws Exception
+     */
+    public void confirmJqiMsg(String messageType) throws Exception {
+        if (isJqiMsgShowing(messageType))
         {
             clickButton(OK);
         }
         else
         {
-            throw new Exception("Permissions editing not successful for row : " + rowNum);
+            throw new Exception("Could not confirm JqiMsg");
         }
+    }
+
+    public void setGroupOnly(String idToFind, String newGroup, boolean confirm) throws Exception
+    {
+        String currentPermission = "";
+        WebElement permissionCheckbox = (new WebDriverWait(driver, 10))
+                .until(ExpectedConditions.elementToBeClickable(
+                        By.id("publicPermission")));
+
+        WebElement groupInput = find(By.id(idToFind));
+
+        currentPermission = permissionCheckbox.getAttribute("checked");
+        if (currentPermission != null) {
+            // clear checkbox
+            // read group field should be enabled
+            click(permissionCheckbox);
+        }
+        sendKeys(groupInput, newGroup);
+
+        clickButton(SAVE);
+
+        // read/writeGroupDiv should have 'has-error' class
+        // confirm here is conditional because it won't
+        // show up if an invalid group has been sent in.
+        if (confirm == true)
+        {
+            confirmJqiMsg(SUCCESSFUL);
+        }
+    }
+
+
+    public String togglePublicAttributeForRow(int rowNum) throws Exception
+    {
+        String currentPermission = "";
+        clickEditIconForFirstRow();
+        WebElement permissionCheckbox = (new WebDriverWait(driver, 10))
+                .until(ExpectedConditions.elementToBeClickable(
+                        By.id("publicPermission")));
+
+        click(permissionCheckbox);
+        currentPermission = permissionCheckbox.getAttribute("checked");
+
+        // This seems to be the only way to get the script to sleep long
+        // enough for the autocomplete ajax call to complete so clicking
+        // the public checkbox then save doesn't result in an error.
+        Thread.sleep(4000);
```

Comment:
The `Thread.sleep()` calls could be problematic and a sign of something wrong.  Could they not wait for an item to appear?